src/draft-service.ts

```diff
@@ -352,8 +333,32 @@ async function updateDraft({
     if (formSubmissionDraftVersion) {
       localDraftsStorage.syncedFormSubmissionDrafts =
         await getFormSubmissionDrafts(draftSubmission.formsAppId, abortSignal)
+      // Remove draft from unsynced incase it was created offline
+      localDraftsStorage.unsyncedDraftSubmissions =
+        localDraftsStorage.unsyncedDraftSubmissions.filter(
+          (unsycnedDraftSubmission) =>
+            unsycnedDraftSubmission.formSubmissionDraftId !==
+            formSubmissionDraftId,
+        )
     } else {
-      localDraftsStorage.unsyncedDraftSubmissions.push(draftSubmission)
+      let updated = false
+      localDraftsStorage.unsyncedDraftSubmissions =
+        localDraftsStorage.unsyncedDraftSubmissions.map(
+          (unsycnedDraftSubmission) => {
```

Comment:
```suggestion
          (unsyncedDraftSubmission) => {
```
Minor spelling mistake